c7n/resources/glacier.py

```diff
@@ -13,23 +13,54 @@
 # limitations under the License.
 from botocore.exceptions import ClientError
 
-from c7n.filters import CrossAccountAccessFilter
+from c7n.actions import ActionRegistry, AutoTagUser
+from c7n.filters import FilterRegistry, CrossAccountAccessFilter
 from c7n.query import QueryResourceManager
 from c7n.manager import resources
-from c7n.utils import local_session
+from c7n.tags import (
+    TagActionFilter, UniversalTag, UniversalUntag, UniversalTagDelayedAction)
+from c7n.utils import get_retry, local_session
 
 
 @resources.register('glacier')
 class Glacier(QueryResourceManager):
 
-    class Meta(object):
+    filter_registry = FilterRegistry('glacier.filters')
+    filter_registry.register('marked-for-op', TagActionFilter)
+
+    action_registry = ActionRegistry('glacier.actions')
+    action_registry.register('auto-tag-user', AutoTagUser)
+    action_registry.register('mark', UniversalTag)
+    action_registry.register('tag', UniversalTag)
+    action_registry.register('mark-for-op', UniversalTagDelayedAction)
+    action_registry.register('remove-tag', UniversalUntag)
+    action_registry.register('unmark', UniversalUntag)
+    action_registry.register('untag', UniversalUntag)
+
+    permissions = ('glacier:ListTagsForVault',)
+    retry = staticmethod(get_retry(('Throttled',)))
+
+    class resource_type(object):
         service = 'glacier'
         enum_spec = ('list_vaults', 'VaultList', None)
         name = "VaultName"
         id = "VaultARN"
         dimension = None
 
-    resource_type = Meta
+    def augment(self, resources):
+        def process_tags(resource):
+            client = local_session(self.session_factory).client('glacier')
```

Comment:
no worries, we can do that as a separate pull request, let me if you have a chance to sync up on this pr.